src/protagonist/DLCS.AWS.Tests/SQS/Models/QueueMessageXTests.cs

```diff
@@ -70,4 +75,59 @@ public void GetMessageContents_ReturnsNull_IfSnsOrigin_AndBodyNotJson()
         // Assert
         contents.Should().BeNull();
     }
+    
+    [Fact] 
+    public void GetMessageContentsWithType_ReturnsObject_WhenCalledWithObject()
+    {
+        // Arrange
+        var asset = new Asset()
+        {
+            Id = new AssetId(1, 99, "foo")
+        };
+        
+        var serialized =  JsonSerializer.Serialize(asset, settings);
+        
+        var queueMessage = new QueueMessage
+        {
+            Body = JsonNode.Parse(serialized)!.AsObject()
+        };
+        
+        // Act
+        var contents = queueMessage.GetMessageContents<Asset>();
+        
+        // Assert
+        contents?.Id.Should().NotBeNull();
+        contents?.Id.Customer.Should().Be(1);
+        contents?.Id.Space.Should().Be(99);
+        contents?.Id.Asset.Should().Be("foo");
+    }
+    
+    [Fact] 
+    public void GetMessageContentsWithType_ThrowsException_WhenCalledWithObjectThatCannotBeDeserialized()
+    {
+        // Arrange
+        var queueMessage = new QueueMessage
+        {
+            Body = new JsonObject { ["id"] = "foo" }
+        };
+
+        // Act and Assert
+        Assert.Throws<JsonException>(() =>queueMessage.GetMessageContents<Asset>());

```

Comment:
Idiomatic way to do this with FluentAssertions is to use an `Action` and assert that

```cs
Action action = () => queueMessage.GetMessageContents<Asset>();
action.Should().Throw<JsonException>(); // or .ThrowExactly<JsonException>()
```